alb.tf

```diff
@@ -90,26 +86,57 @@ resource "aws_alb_listener_rule" "public_80" {
 }
 
 data "aws_lb" "private" {
-  name = "private"
+  count = var.alb_attach_private_target_group ? 1 : 0
+  name  = "private"
 }
 
 data "aws_lb_listener" "private" {
-  load_balancer_arn = data.aws_lb.private.arn
+  count             = var.alb_attach_private_target_group ? 1 : 0
+  load_balancer_arn = data.aws_lb.private[count.index].arn
   port              = 80
 }
 
+resource "aws_alb_target_group" "private" {
+  count = var.alb_attach_private_target_group ? 1 : 0
+
+  deregistration_delay = 5
+  name                 = "${var.service_name}-private"
+  port                 = var.container_port
+  protocol             = "HTTP"
+  tags                 = local.default_tags
+  target_type          = "ip"
+  vpc_id               = data.aws_vpc.selected.id
+
+  dynamic "health_check" {
```

Comment:
question: can you directly use `var.health_check` instead?

Reply:
this needs to be a block, so direct assignment won't work